src/Flow.jl

```diff
@@ -66,10 +66,16 @@ using EllipsisNotation
 This function adds a uniform acceleration field `g` at time `t` to `r`.
 If `g ≠ nothing`, then `g(i,t)=dUᵢ/dt`.
 """
-accelerate!(r,t,g) = for i ∈ 1:last(size(r))
+accelerate!(r,t,g::Function,::Tuple) = for i ∈ 1:last(size(r))
     r[..,i] .+= g(i,t)
 end
-accelerate!(r,t,::Nothing) = nothing
+accelerate!(r,t,::Nothing,U::Function) = for i ∈ 1:last(size(r))
+    r[..,i] .+= ForwardDiff.derivative(τ->U(i,τ),t)
+end
+accelerate(r,t,::Function,::Function) = for i ∈ 1:last(size(r))
+    r[..,i] .+= g(i,t) + ForwardDiff.derivative(τ->U(i,τ),t)
```

Comment:
`U` seems to not be in the scope. Also this is not covered by tests which is why it is not being picked up as a test error.

Reply:
Neither is `g` actually. Luckily I didn't test it ;)

Reply:
And the name is `accelerate()`, not `accelerate!()`...